examples/dtls-echo/dtls-client.c

```diff
@@ -23,56 +24,38 @@
 #include <stdio.h>
 #include <inttypes.h>
 
-#include "net/gnrc.h"
-#include "net/gnrc/ipv6.h"
-#include "net/gnrc/netif.h"
-#include "net/gnrc/netif/hdr.h"
-#include "net/gnrc/udp.h"
-#include "net/gnrc/pktdump.h"
-#include "timex.h"
-#include "utlist.h"
-#include "xtimer.h"
-
-#define ENABLE_DEBUG (0)
-#include "debug.h"
+#include "net/sock/udp.h"
 
 /* TinyDTLS */
-#include "tinydtls.h"
 #include "dtls_debug.h"
 #include "dtls.h"
-#include "global.h"
-
-
 
-/* TODO: Remove the UNUSED_PARAM from TinyDTLS' stack? */
-#ifdef __GNUC__
-#define UNUSED_PARAM __attribute__((unused))
-#else
-#define UNUSED_PARAM
-#endif /* __GNUC__ */
+#define ENABLE_DEBUG (0)
+#include "debug.h"
 
 #ifdef DTLS_PSK
 
 #define PSK_DEFAULT_IDENTITY "Client_identity"
-#define PSK_DEFAULT_KEY      "secretPSK"
-#define PSK_OPTIONS          "i:k:"
+#define PSK_DEFAULT_KEY "secretPSK"
```

Comment:
Those defines have to be set on both sides (client and server) and should the same. Here there's a problem: 1. a define is used in the client, 2. they are hardcoded in the server code. 

I propose to have them configurable at build time in the Makefile of the application and defined globally:
- in the Makefile:
```mk
PSK_DEFAULT_IDENTITY ?= "Client_identity"
PSK_DEFAULT_KEY ?= "secretPSK"
...
CFLAGS += -DPSK_DEFAULT_IDENTITY=$(PSK_DEFAULT_IDENTITY) \
          -DPSK_DEFAULT_KEY=$(PSK_DEFAULT_KEY)
```
- in the c files of client and server (or maybe in a shared header file ?):
```c
#ifndef PSK_DEFAULT_IDENTITY
#define PSK_DEFAULT_IDENTITY "Client_identity"
#endif

#ifndef PSK_DEFAULT_KEY
#define PSK_DEFAULT_KEY "secretPSK"
#endif
```

Reply:
They were remaining from the original examples for POSIX, but yeah, it's more practical to make the changes. At least for this example.

Reply:
mmm, Ok. Maybe a header for the keys is a better idea. Otherwise, the private and public keys for the RPK should be provided in a similar fashion in the Makefile.